core/controller/src/main/scala/org/apache/openwhisk/core/controller/actions/PrimitiveActions.scala

```diff
@@ -599,7 +599,8 @@ protected[actions] trait PrimitiveActions {
       case Right(activation) => result.trySuccess(Right(activation))
       case _ if (controllerActivationConfig.pollingFromDb) =>
         pollActivation(docid, context, result, i => 1.seconds + (2.seconds * i), maxRetries = 4)
-      case _ =>
+      case Left(activationId) =>
+        result.trySuccess(Left(activationId)) // complete the future immediately if it's configured to not poll db for blocking activations
```

Comment:
hmm - convince me you actually tested this change.

Reply:
I created an issue #4231

Reply:
This change is sound. This block definitely should resolve the promise on every branch in some way or the other. If database polling is disabled, the promise is essentially blackholed.

A RecordTooLargeException results indeed in the fixed `Left` case, see https://github.com/apache/incubator-openwhisk/blob/7ff15502949fa21ab5621825204397ad9eb6dfe5/core/invoker/src/main/scala/org/apache/openwhisk/core/invoker/InvokerReactive.scala#L168-L171